pkg/ottl/expression.go

```diff
@@ -245,6 +246,55 @@ func (g StandardFloatGetter[K]) Get(ctx context.Context, tCtx K) (float64, error
 	}
 }
 
+// FunctionGetter is a Getter that must return an ExprFunc.
+type FunctionGetter[K any] interface {
+	Get(ctx context.Context, iArgs Arguments) (Expr[K], error)
+}
+
+// StandardFunctionGetter is a basic implementation of FunctionGetter
+type StandardFunctionGetter[K any] struct {
+	fCtx FunctionContext
+	fact Factory[K]
+}
+
+// Get retrieves an ExprFunc value.
+// If the value is not an ExprFunc a new Error is returned.
+// If there is an error getting the value it will be returned.
+func (g StandardFunctionGetter[K]) Get(_ context.Context, iArgs Arguments) (Expr[K], error) {
+	if g.fact == nil {
+		return Expr[K]{}, fmt.Errorf("undefined function")
+	}
+	fArgs := g.fact.CreateDefaultArguments()
+	fArgsVal := reflect.ValueOf(fArgs).Elem()
+	iArgsVal := reflect.ValueOf(iArgs).Elem()
+	if fArgsVal.NumField() != iArgsVal.NumField() {
+		return Expr[K]{}, fmt.Errorf("incorrect number of arguments. Expected: %d Received: %d", fArgsVal.NumField(), iArgsVal.NumField())
+	}
+	for i := 0; i < fArgsVal.NumField(); i++ {
+		field := iArgsVal.Field(i)
+		fieldTag := fArgsVal.Type().Field(i).Tag.Get("ottlarg")
+		if fieldTag == "" {
+			return Expr[K]{}, fmt.Errorf("no `ottlarg` struct tag on Arguments field %q", fArgsVal.Type().Field(i).Name)
+		}
+		argNum, err := strconv.Atoi(fieldTag)
+		if err != nil {
+			return Expr[K]{}, fmt.Errorf("ottlarg struct tag on field %q is not a valid integer: %w", fArgsVal.Type().Field(i).Name, err)
+		}
+		if argNum < 0 || argNum >= iArgsVal.NumField() {
+			return Expr[K]{}, fmt.Errorf("ottlarg struct tag on field %q has value %d, but must be between 0 and %d", fArgsVal.Type().Field(i).Name, argNum, iArgsVal.NumField())
+		}
+		if field.Type() != fArgsVal.Field(i).Type() {
+			return Expr[K]{}, fmt.Errorf("incorrect type for argument %d. Expected: %v Received: %v", i, fArgsVal.Field(i).Type(), field.Type())
+		}
+		fArgsVal.Field(i).Set(field)
```

Comment:
If all this logic is really similar to the logic in `buildArgs` it would be nice to break it out into a helper function and reuse it.

Reply:
yes there is some similarity, ok I'll work on putting together a helper function.

Reply:
Added a helper function for argument error checking